lib/improver/tests/threshold/test_BasicThreshold.py

```diff
@@ -177,6 +213,107 @@ def test_threshold_fuzzy(self):
         expected_result_array[0][0][2][2] = 1.0/3.0
         self.assertArrayAlmostEqual(result.data, expected_result_array)
 
+    def test_threshold_fuzzybounds(self):
+        """Test when a point is in the fuzzy threshold area."""
+        bounds = (0.6 * self.fuzzy_factor, 0.6 * (2. - self.fuzzy_factor))
+        plugin = Threshold(0.6, fuzzy_bounds=bounds)
+        result = plugin.process(self.cube)
+        expected_result_array = np.zeros_like(self.cube.data).reshape(
+            1, 1, 5, 5)
+        expected_result_array[0][0][2][2] = 1.0/3.0
+        self.assertArrayAlmostEqual(result.data, expected_result_array)
+
+    def test_threshold_boundingzero(self):
+        """Test fuzzy threshold of zero."""
+        bounds = (-1.0, 1.0)
+        plugin = Threshold(0.0, fuzzy_bounds=bounds)
+        result = plugin.process(self.cube)
+        expected_result_array = np.full_like(
+            self.cube.data, fill_value=0.5).reshape(1, 1, 5, 5)
+        expected_result_array[0][0][2][2] = 0.75
+        self.assertArrayAlmostEqual(result.data, expected_result_array)
+
+    def test_threshold_boundingzero_above(self):
+        """Test fuzzy threshold of zero."""
+        bounds = (-0.1, 0.1)
+        plugin = Threshold(0.0, fuzzy_bounds=bounds)
+        result = plugin.process(self.cube)
+        expected_result_array = np.full_like(
+            self.cube.data, fill_value=0.5).reshape(1, 1, 5, 5)
+        expected_result_array[0][0][2][2] = 1.
+        self.assertArrayAlmostEqual(result.data, expected_result_array)
+
+    def test_threshold_boundingbelowzero(self):
+        """Test fuzzy threshold of below-zero."""
+        bounds = (-1.0, 1.0)
+        plugin = Threshold(0.0, fuzzy_bounds=bounds, below_thresh_ok=True)
+        result = plugin.process(self.cube)
+        expected_result_array = np.full_like(
+            self.cube.data, fill_value=0.5).reshape(1, 1, 5, 5)
+        expected_result_array[0][0][2][2] = 0.25
+        self.assertArrayAlmostEqual(result.data, expected_result_array)
+
+    def test_threshold_assymetric_bounds_below(self):
```

Comment:
All your "asymmetric " titles are spelt incorrectly; double m not double s.

Reply:
Fixed - thanks.